angular/projects/admin-nrpti/src/app/mines/mines-records-list-resolver.ts

```diff
@@ -0,0 +1,138 @@
+
+    let schemaList = [
+      'Order',
+      'Inspection',
+      'Certificate',
+      'Permit',
+      'SelfReport',
+      'Agreement',
+      'RestorativeJustice',
+      'Ticket',
+      'AdministrativePenalty',
+      'AdministrativeSanction',
+      'Warning',
+      'ConstructionPlan',
+      'ManagementPlan',
+      'CourtConviction'
+    ];
+
+    if (params.activityType) {
+      schemaList = params.activityType.split(',');
+    }
+
+    let keywords = '';
+    if (params.keywords) {
+      keywords = params.keywords;
+    }
+    let subset = [];
+    if (params.subset) {
+      subset = params.subset.split(',');
+    }
+
+    const and = {};
+    const or = {};
+    const nor = {};
+
+    if (params.dateRangeFromFilter) {
+      or['dateRangeFromFilterdateIssued'] = params.dateRangeFromFilter;
+    }
+
+    if (params.dateRangeToFilter) {
+      or['dateRangeToFilterdateIssued'] = params.dateRangeToFilter;
+    }
+
+    if (params.issuedToCompany && params.issuedToIndividual) {
+      or['issuedTo.type'] = 'Company,Individual,IndividualCombined';
+    } else if (params.issuedToCompany) {
+      or['issuedTo.type'] = 'Company';
+    } else if (params.issuedToIndividual) {
+      or['issuedTo.type'] = 'Individual,IndividualCombined';
+    }
+
+    if (params.agency) {
+      or['issuingAgency'] = params.agency;
+    }
+
+    if (params.act) {
+      or['legislation.act'] = params.act;
+    }
+
+    if (params.regulation) {
+      or['legislation.regulation'] = params.regulation;
+    }
+
+    if (params.sourceSystemRef) {
+      or['sourceSystemRef'] = params.sourceSystemRef;
+    }
+
+    if (params.hasDocuments) {
+      or['hasDocuments'] = params.hasDocuments;
+    }
+
+    if (params.projects) {
+      const projectIds = [];
+
+      if (params.projects.includes('lngCanada')) {
+        projectIds.push(EpicProjectIds.lngCanadaId);
+      }
+
+      if (params.projects.includes('coastalGaslink')) {
+        projectIds.push(EpicProjectIds.coastalGaslinkId);
+      }
+
+      if (params.projects.includes('otherProjects')) {
+        if (projectIds.length === 0) {
+          // Selecting only Other should return all projects EXCEPT for LNG Canada and Coastal Gaslink
+          nor['_epicProjectId'] = `${EpicProjectIds.lngCanadaId},${EpicProjectIds.coastalGaslinkId}`;
+        } else if (projectIds.length === 1) {
+          if (projectIds[0] === EpicProjectIds.lngCanadaId) {
+            // Other + LNG Canada is equivalent to NOT Coastal Gaslink
+            nor['_epicProjectId'] = EpicProjectIds.coastalGaslinkId;
+          } else {
+            // Other + Coastal Gaslink is equivalent to NOT LNG Canada
+            nor['_epicProjectId'] = EpicProjectIds.lngCanadaId;
+          }
+        }
+      } else if (projectIds.length) {
+        or['_epicProjectId'] = projectIds.join(',');
+      }
+    }
+
+    if (params.isNrcedPublished) {
+      or['isNrcedPublished'] = params.isNrcedPublished;
+    }
+    if (params.isLngPublished) {
```

Comment:
Do we need to handle `isBcmiPublished`?

Reply:
These search filters are not the final ones that this page will have.  I just dropped the existing ones in, because it was an easy way to get the filters kick started, and plumb in the related code that will likely be similar when the real filters get added.  I imagine most of the common filter logic will remain, it will just be a matter of adding some new ones for collections, and maybe removing some that dont matter to mines or bcmi.  

So I think for now this is okay? And will get updated and/or removed in the other ticket that deals with the filters more closely.

Reply:
Sounds good.